### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by Bupilipili**
### In this branch

- Added Redux Toolkit (npm install react-redux @reduxjs/toolkit).
- Created a directory that will contain all your Redux logic (/src/redux)
- Configured a Redux store (/src/redux/store.js)
- Defined a slice of state for books that:
 Includes an array of books (initial state: empty array)
 Includes a reducer that adds a book
 Includes a reducer that removes a book
- Defined a slice of state for categories that:
 Includes an array of categories (initial state: empty array)
 Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)
- Structured the application files using a "feature folder" approach and used the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for the Redux files. The files/folder structure look like this:
/src
    |---/components
    |---/redux
        |--- /books
                | booksSlice.js
        |--- /categories
                | categoriesSlice.js
        | store.js
    | index.js
- Did not change React components.

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  categories: [{ status: 'Under construction' }],
+};
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {},
+});
+
+export default categoriesSlice.reducer;
```
Comment (rloterh) [inline src/redux/categories/categoriesSlice.js]: - [x] Let's improve the construction of the _CategoriesSlice_ by ensuring the following:
   - Creates an initialState with a state value categories that contains an empty array
   - Creates 1 reducer, checkStatus
   - Exports by checkStatus as an action

And then the CategoriesSlice should be similar to:

```javascript
// ./src/redux/categories/categoriesSlice.js

import { createSlice } from "@reduxjs/toolkit"

export const categoriesSlice = createSlice({
  name: "categories",
  initialState: {
    categories: [],
  },
  reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
})

export const { checkStatus } = categoriesSlice.actions

export default categoriesSlice.reducer
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Erick, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🚦 🌴  🏃 🌳 🏡 🏘️ 🌲 

## Highlights 👍  
- Good job so far on state management!
- Good ReadMe!

## Required Changes ♻️
- [x] Deployment link in your ReadMe is broken, kindly endeavor to fix it.

_Kindly check other inline comments under the review._


Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (Bupilipili) [reply src/redux/categories/categoriesSlice.js]: Thanks for the good insight 😃

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi @Bupilipili 👋 , 

Great job implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 🎯 
✔️ Descriptive PR
✔️ Added actions and reducers 📌 
✔️ No linter errors 🟢

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [x] There appears to be an eslint problem that affects the correct compilation of your application. Please look into rectifying this issue 🙏 
image


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Bupilipili** [comment]
OK am on it... 👏

**Pull request merged**